src/bindings/c/src/ov_infer_request.cpp

```diff
@@ -289,13 +289,13 @@ ov_status_e ov_infer_request_wait_for(ov_infer_request_t* infer_request, const i
     if (!infer_request) {
         return ov_status_e::INVALID_C_PARAM;
     }
-
+    bool ret = true;
     try {
-        infer_request->object->wait_for(std::chrono::milliseconds(timeout));
+        ret = infer_request->object->wait_for(std::chrono::milliseconds(timeout));
     }
     CATCH_OV_EXCEPTIONS
 
-    return ov_status_e::OK;
+    return ret ? ov_status_e::OK : ov_status_e::UNEXPECTED;
```

Comment:
agree that [RESULT_NOT_READY](https://github.com/openvinotoolkit/openvino/blob/8c09df0f3e83287aea2fc9fc7eafd0867e1e82f1/src/bindings/c/include/openvino/c/ov_common.h#L151) is more clear

Reply:
ov::InferRequest::wait_for() API returns bool value:
https://github.com/openvinotoolkit/openvino/blob/dcf87efe51dc9710ba71a39916c74ae935119597/src/inference/include/openvino/runtime/infer_request.hpp#L287-L294

If there is any exception, C API will catch it and throw exception rather than return an error code in line 298.

If there is no exception and return value is `false`,  it means `std::future_status` is not ready or not valid, it makes sense to set `RESULT_NOT_READY` for C API return value.